openstudiocore/src/utilities/core/test/ApplicationPathHelpers_GTest.cpp

```diff
@@ -51,3 +113,204 @@ TEST(ApplicationPathHelpers, Strings)
   EXPECT_TRUE(exists(energyplusDirectory));
   EXPECT_TRUE(exists(energyplusExecutable));
 }
+
+
+
+// This is really in PathHelpers but used by ApplicationPathHelpers
+TEST(ApplicationPathHelpers, findInSystemPath) {
+
+  openstudio::path dummy_dir = getApplicationBuildDirectory() / toPath("Testing/Temporary");
+  openstudio::filesystem::create_directories(dummy_dir);
+  openstudio::path dummy_file_path = dummy_dir / "dummy_file";
+
+  boost::filesystem::ofstream outfile(dummy_file_path);
+  outfile.close();
+
+  // This should fail
+  openstudio::path absolute_path_to_dummy = findInSystemPath(dummy_file_path.filename());
+  EXPECT_FALSE(exists(absolute_path_to_dummy));
+
+  // Append the containing dir to the PATH
+  auto current_path = std::getenv("PATH");
+  std::string new_path = toString(dummy_dir) + getPlatformDelimiterForPath() + current_path;
+  setenv("PATH", new_path.c_str(), 1);
+  setenv("PathWhenSymlinkInPathUnixOnly_Setup", "true", 1);
+
+  // Locate the file with only its name
+  absolute_path_to_dummy = findInSystemPath(dummy_file_path.filename());
+  EXPECT_TRUE(exists(absolute_path_to_dummy)) << "Tried to find " << dummy_file_path.filename() << "in path: " << current_path;
+  EXPECT_EQ(dummy_file_path, absolute_path_to_dummy);
+
+  // Put it back
+  setenv("PATH", current_path, 1);
+  boost::filesystem::remove(dummy_file_path);
+
+}
+
+// From PathHelpers too. Ensure that completeAndNormalize keeps resolving symlinks until found
+TEST(ApplicationPathHelpers, completeAndNormalizeMultipleSymlinks) {
+
+  if (!IsElevated()) {
+   SKIP(completeAndNormalizeMultipleSymlinks);
+  }
+
+  std::vector<openstudio::path> toClean;
+  openstudio::path path_dir = getApplicationBuildDirectory() / toPath("Testing/Temporary");
+  openstudio::path path_subdir = path_dir / toPath("SubDir");
+
+  openstudio::filesystem::create_directories(path_subdir);
+
+  openstudio::path ori_path = path_dir / "completeAndNormalize_oriFile";
+  toClean.push_back(ori_path);
+
+  // 'touch'
+  boost::filesystem::ofstream outfile(ori_path);
+  outfile.close();
+
+  openstudio::path prev_path = ori_path;
+
+  // First one, same directory, absolute
+  openstudio::path symlink_path = path_dir / toPath("1.symlink_absolute_same_dir");
+  openstudio::path pointsTo = prev_path;
+  boost::filesystem::create_symlink(pointsTo, symlink_path);
```

Comment:
@jmarrec does this work on windows?

Reply:
cf: https://github.com/NREL/OpenStudio/pull/3497/commits/40c4747612d40e967e4de4dd299b676dd37ad598

"MUST BE RUN AS ADMIN CMD.EXE". That's why I do this:

```
TEST(ApplicationPathHelpers, PathWhenSymlinkInPath_Setup) {

  if (!IsElevated()) {
    SKIP(completeAndNormalizeMultipleSymlinks);
  }

  [...]
```

So on windows it'll typically skip it. But if I run CMD.EXE as an admin, it'll run the tests